goldmane/pkg/daemon/config.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/goldmane/pkg/aggregator"
+	"github.com/projectcalico/calico/goldmane/pkg/internal/utils"
+)
+
+type goldmaneFileConfig struct {
```

Comment:
Maybe this is preference, but to me configuration is easier to understand when it's all together so I would advocate for adding this logic to the config object that we already have.

I don't think we need multiple config structures just because there's multiple sources.

Reply:
I wasn't thinking this should also be configurable via env variables as well, including it in the config struct doesn't mean that.

I don't think the config struct should represent what the source of the configuration IMO, it's just everything that's needed to configure goldmane put into a struct that so the dameon can configure it. I don't feel there's a need to have multiple config structs to configure goldmane.

This said, leave it if you don't agree or even if you just don't want to do it, this is probably close to splitting hairs and in general preference (especially when there's just two structs).